draft/0005-impersonation.md

```diff
@@ -0,0 +1,279 @@
+    m9maWxlIiwibmFtZSI6IkpvaG4gRG9lIiwiaWF0IjoxNTE2MjM5MDIyL
+    CJhY3QiOnsic3ViIjoiNDU2In19.H-X4W0mu-lJae27_rvsHAx5LpzsH
+    TDYozkWDWzBrMoM",
+  "scope":"openid offline_access profile",
+  "issued_token_type":
+    "urn:ietf:params:oauth:token-type:access_token",
+  "token_type":"Bearer",
+  "expires_in":3600
+}
+```
+
+The payload of the access token:
+
+```json
+{
+  "sub": "123",
+  "scope": "openid offline_access profile",
+  "client_id": "client1",
+  "name": "John Doe",
+  "iat": 1516239022
+}
+```
+
+## Subject token
+
+The `subject_token` parameter in the token exchange request represents the identity of the party on behalf of whom the request is being made. The `subject_token` is a special one-time access token that is issued to the authorized user by Logto's management API. The `subject_token` is an opaque token that is not intended to be used directly by the client application. Instead, the client application exchanges the `subject_token` for a new access token that represents the identity of the impersonated user. Because the `subject_token` is only consumed by Logto as the authorization server, it does not need to be a JWT.
+
+### Issuing a subject token
+
+#### Request
+
+The client makes a request to the tenant's management API to issue a subject token. The following parameters are included in the HTTP request entity-body using the `application/json` format.
+
+1. `userId`: REQUIRED. The ID of the user on behalf of whom the request is being made.
+
+#### Response
+
+If the request to issue a subject token is successful, the tenant's management API returns the subject token in the response.
+
+1. `subject_token`: REQUIRED. The subject token that represents the identity of the user.
+2. `expires_in`: REQUIRED. The lifetime in seconds of the subject token.
+
+#### Example subject token issuance
+
+The following example demonstrates a request to issue a subject token and the response. This request is made by a machine-to-machine application that has the necessary permissions to issue subject tokens.
+
+```
+POST /api/security/subject-tokens HTTP/1.1
+Host: example.logto.app
+Authorization: Bearer eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9...
+Content-Type: application/json
+
+{
+  "userId":"123456",
+  "expiresIn":600
+}
+```
+
+Logto validates the `userId` and permissions to issue a subject token. If the request is valid, Logto generates a new subject token that represents the identity of the user and returns it in the response. Here is an example of the response:
+
+```
+HTTP/1.1 200 OK
+Content-Type: application/json
+
+{
+  "subjectToken":"accVkjcJyb4BWCxGsndESCJQbdFMogUC5PbRDqceLTC",
+  "expiresIn":600
+}
+```
+
+## Additional claims in the access token
+
+### `act`
+
+When `actor_token` presents in the request of token exchange, the access token issued by Logto will contain an additional `act` claim that represents the identity of the impersonated user. The `act` claim is an object that contains the following fields.
```

Comment:
1. how to get an `actor_token`?
2. for security reasons, should we mark this as required? (not sure)

Reply:
1. The requirement of `actor_token` is added in the "Request" section. This is actully the access token of the support engineer (in this example).
2. According to the RFC, this is not required.